(afterfixed)homework2.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+// I had fixed errors not only about SOLID principle but also other errors.
+
+interface workerInterface
+{
+    public  function work();
+}
+
+interface sleepInterface
+{
+    public  function  sleep();
+}
+
+interface hasPowerInterface
+{
+    public  function  hasPower();
+}
+
+class HumanWorker implements workerInterface
+{
+    public  function work()
+    {
+        var_dump('works');
+    }
+
+    public  function  sleep()
+    {
+        var_dump('sleep');
+    }
+
+}
+
+class RobotWorker extends HumanWorker
```

Comment:
RobotWorker can not  be substitutable by HumanWorer beucase RobotWoker.work must check power before working as well as RobotWoker.sleep is not used.
The codes violates Liskov Substitution Principle.

We should remove extends HumanWorker in here.